main.cpp

```diff
@@ -62,11 +64,25 @@ std::unique_ptr<GenericUI> CreateUI(int argc, char** argv) {
   return std::make_unique<CommandLineUI>();
 }
 
-}  // namespace
+std::unique_ptr<GenericUI> CreateUI(int argc, char** argv) {
+  return std::make_unique<WindowsUI>();
+}
+
+const bool Type_of_UI = ReadCommandLine(argc, argv);
 
+}  // namespace
 
 int main(int argc, char** argv) {
   auto ifg = CreateUI(argc, argv);
   ifg->Run();
   return 0;
+}
+//-----------------------------------Functions------------------------------//
+
+const bool Type_of_UI = ReadCommandLine(argc, argv);
+if (Type_of_UI) {
```

Comment:
local variables should be named in lower case: ui_type is a good and concise name.